scalafix-rules/src/main/scala/scalafix/internal/pc/Embedded.scala

```diff
@@ -0,0 +1,112 @@
+package scalafix.internal.pc
+
+import java.net.URLClassLoader
+import java.util.ServiceLoader
+
+import scala.jdk.CollectionConverters._
+
+import scala.meta.pc.PresentationCompiler
+
+import coursierapi.Dependency
+import coursierapi.Fetch
+import coursierapi.MavenRepository
+import scala.collection.concurrent.TrieMap
+
+object Embedded {
+
+  private val presentationCompilers: TrieMap[String, URLClassLoader] =
+    TrieMap.empty
+
+  def presentationCompiler(
+      scalaVersion: String
+  ): PresentationCompiler = {
+    val classloader = presentationCompilers.getOrElseUpdate(
+      scalaVersion,
+      newPresentationCompilerClassLoader(scalaVersion)
+    )
+    val presentationCompilerClassname =
+      if (supportPresentationCompilerInDotty(scalaVersion)) {
+        "dotty.tools.pc.ScalaPresentationCompiler"
+      } else {
+        "scala.meta.pc.ScalaPresentationCompiler"
+      }
```

Comment:
This is needed only if running against Scala <3.4.0 since that's when mtags when upstreamed with a different name if I understand well. I would rather fail and ask for users to upgrade to LTS - I believe it's a fair thing to ask - or use cli_2x if sources are Scala 2. That, together with the inability to target Scala 3 sources from rules_2 discussed above would remove the need for mtags all together right?

Reply:
I can remove it, but I don't think removing the dynamic fallback is something we should do. I will add a config option

Reply:
I am removing mtags and direct references to mtags-interfaces for now, keeping dynamic loading of the Scala 3 PC behind a flag